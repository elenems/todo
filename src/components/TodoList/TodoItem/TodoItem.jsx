import React from "react";
import IconButton from "@material-ui/core/IconButton";
import { makeStyles } from "@material-ui/core/styles";
import { MdRemoveCircleOutline } from "react-icons/md";
import { MdEdit } from "react-icons/md";
import Box from "@material-ui/core/Box";
import { connect } from "react-redux";
import * as Actions from "../../../store/actions/todoActions";
import Tooltip from "@material-ui/core/Tooltip";
import Typography from "@material-ui/core/Typography";
import { withRouter } from "react-router-dom";
const useStyles = makeStyles({
  root: {
    margin: "0px 0px 10px 0px",
    padding: "12px",
    fontSize: "26px",
    border: "none"
  }
});

function TodoItem(props) {
  const todo = props.todo;
  const classes = useStyles();

  const moveToEditTodo = () => {
    const todoID = props.todo.id;
    props.history.push(`/edit/${todoID}`);
  };

  const removeTodo = e => {
    const id = todo.id;
    const element = e.target.closest(".todo-item");
    element.className += ' hide';
    setTimeout(() => {
      props.removeTodo(id);
    }, 400);
  };

  const toggleTodoStatus = e => {
    props.toggleTodoStatus(props.todo.id);
    const element = e.target;
    const textDecoration =
      element.style.textDecoration === "line-through" ? "none" : "line-through";
    element.style.textDecoration = textDecoration;
  };

  const textStyle = props.todo.isChecked
    ? { textDecoration: "line-through" }
    : { textDecoration: "none" };

  return (
    <Box
      className="todo-item reveal"
      display="flex"
      alignItems="baseline"
      justifyContent="space-between"
    >
      <Typography
        style={textStyle}
        onClick={toggleTodoStatus}
        component="p"
        variant="h5"
      >
        {todo.title}
      </Typography>
      <Box>
        <Tooltip title="Edit">
          <IconButton
            onClick={moveToEditTodo}
            color="primary"
            aria-label="edit"
            classes={{ root: classes.root }}
          >
            <MdEdit />
          </IconButton>
        </Tooltip>
        <Tooltip title="Delete">
          <IconButton
            onClick={removeTodo}
            color="secondary"
            aria-label="delete"
            classes={{ root: classes.root }}
          >
            <MdRemoveCircleOutline />
          </IconButton>
        </Tooltip>
      </Box>
    </Box>
  );
}

const mapDispatchToProps = dispatch => {
  return {
    removeTodo: id => dispatch(Actions.removeTodo(id)),
    toggleTodoStatus: id => dispatch(Actions.toggleTodoStatus(id))
  };
};

export default withRouter(
  connect(
    null,
    mapDispatchToProps
  )(TodoItem)
);
