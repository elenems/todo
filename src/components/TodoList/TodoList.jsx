import React, { Component } from "react";
import TodoItem from "./TodoItem";
import { connect } from "react-redux";
import Box from "@material-ui/core/Box";
import Typography from "@material-ui/core/Typography";

class TodoList extends Component {
  render() {
    const todos = this.props.todos;
    return (
      <Box style={{ marginTop: "20px" }}>
        {todos.length ? (
          todos.map(todo => {
            return <TodoItem key={todo.id} todo={todo} />;
          })
        ) : (
          <Typography component="span" variant="h6" gutterBottom>
            Add your first todo
          </Typography>
        )}
      </Box>
    );
  }
}

const mapStateToProps = state => {
  return {
    todos: state.todo.todos
  };
};

export const TodoListTest = TodoList;

export default connect(mapStateToProps)(TodoList);
