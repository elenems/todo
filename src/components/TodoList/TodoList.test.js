import React from "react";
import { shallow } from "enzyme";
import TodoList from "../TodoList";
import {TodoListTest} from './TodoList.jsx';
import configureMockStore from "redux-mock-store";
import TodoItem from './TodoItem';
describe("TodoList", () => {
  const mockStore = configureMockStore();
  const initialState = {
    todo: {
      todos: [],
      error: "",
      message: ""
    }
  };
  it("Matches snapshot", () => {
    const store = mockStore(initialState);
    const wrapper = shallow(<TodoList store={store} />).dive();
    expect(wrapper).toMatchSnapshot();
  });

  it("Renders TodoItem if passed array has content", () => {
    const todos = [{ id: 1, title: "Todo title" }, {id:2, title:"Some title 2"}];
    const wrapper = shallow(<TodoListTest todos={todos} />);
    expect(wrapper.find(TodoItem).length).toBeGreaterThan(0);
  });
});
