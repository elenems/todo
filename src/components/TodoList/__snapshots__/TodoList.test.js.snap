// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList Matches snapshot 1`] = `
<TodoList
  dispatch={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  todos={Array []}
/>
`;
